src/researches/buildKeywordForms.js

```diff
@@ -41,67 +42,97 @@ const getVariationsApostropheInArray = function( forms ) {
 	} ) ).filter( Boolean );
 };
 
+const filterFunctionWords = function( array, language ) {
+	if ( isUndefined( language ) || language === "" ) {
+		language = "en";
+	}
+
+	const functionWords = getFunctionWords[ language ];
+
+	if ( array.length > 1 && ! ( isUndefined( functionWords ) ) ) {
+		const arrayFiltered = filter( array, function( word ) {
+			return ( ! includes( functionWords.all, word.trim().toLocaleLowerCase() ) );
+		} );
+
+		if ( arrayFiltered.length > 0 ) {
+			return arrayFiltered;
+		}
+	}
+
+	return array;
+};
+
 /**
  * Analyzes the focus keyword string. Checks if morphology is requested or if the user wants to match exact string.
  * If morphology is required the module builds all word forms for all content words (prepositions, articles, conjunctions).
  *
- * @param {Object} paper The paper to analyze keyword forms for.
+ * @param {string} keyphrase The keyphrase of the paper (or a synonym phrase) to get forms for.
+ * @param {string} language The language to use for morphological analyzer and for function words.
+ * @param {boolean} morphologyRequired Whether the morphological analysis is required (i.e, Premium vs. Free).
  *
  * @returns {Array} Array of all forms to be searched for keyword-based assessments.
  */
-module.exports = function( paper ) {
-	let keyword = paper.getKeyword();
-	const language = getLanguage( paper.getLocale() );
-	const getForms = getFormsForLanguage[ language ];
-
-	/*
-	 * If the language is not yet supported with respect to morphological analysis, return keyword itself,
-	 * including variations of the apostrophe in case it is present.
-	 */
-	if ( isUndefined( getForms ) ) {
-		return unique( [].concat( keyword, getVariationsApostrophe( keyword ) ) );
+const buildForms = function( keyphrase, language, morphologyRequired = false ) {
+	if ( isUndefined( keyphrase ) || keyphrase === "" ) {
+		return [];
 	}
 
+	// If the keyphrase is embedded in double quotation marks, return keyword itself, without outer-most quotation marks.
```

Comment:
I think it would be good to also mention the apostrophe variations that are added.